chain/network/src/network_protocol/proto_conv/util.rs

```diff
@@ -0,0 +1,41 @@
+/// Proto conversion utilities.
+use protobuf::MessageField as MF;
+
+#[derive(thiserror::Error, Debug)]
+#[error("[{idx}]: {source}")]
+pub struct ParseVecError<E> {
+    idx: usize,
+    #[source]
+    source: E,
+}
+
+pub fn try_from_vec<'a, X, Y: TryFrom<&'a X>>(
+    xs: &'a Vec<X>,
```

Comment:
Unless I am missing something, this looks like it want to be:
```suggestion
    xs: &'a [X],
```
(There are very few reasons to ever use `&Vec<_>` in a parameter, slices are just more versatile.)

Reply:
done